### PR TITLE
[GraphBolt] Reorganzie graphbolt module

### PR DESCRIPTION
**Pull request by peizhou001**
## Description
This PR rename the folder name of graphbolt and remove the currenltly unsused files inside it.

## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/__init__.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
 """GraphBolt"""
 
+from .graph_storage import *
```
Comment (Rhett-Ying) [inline python/dgl/graphbolt/__init__.py:3]: does `graph_storage` exist?

Comment (peizhou001) [reply python/dgl/graphbolt/__init__.py:3]: For later PR usage

Comment (Rhett-Ying) [reply python/dgl/graphbolt/__init__.py:3]: then it's better to add this at that time.

**Review by Rhett-Ying — APPROVED** [review]
overall LGTM. How about replace `base.py` with `__init__.py` to mark it as a module?

File: python/dgl/graphbolt/__init__.py (hunk 1 above)
Comment (peizhou001) [reply python/dgl/graphbolt/__init__.py:3]: Both would be ok, according to the comment in #5700 , we move the `graphbolt` level changes in this PR.

File: python/dgl/__init__.py [hunk 2]
```diff
@@ -48,7 +47,6 @@
     set_node_lazy_features,
     set_src_lazy_features,
 )
-from .dataloading2 import *
```
Comment (Rhett-Ying) [inline python/dgl/__init__.py:51]: this results in CI failure, I think.

File: python/dgl/graphbolt/__init__.py (hunk 1 above)
Comment (frozenbugs) [reply python/dgl/graphbolt/__init__.py:3]: Only add code when the code is valid, if the graph_storage does not even exist, why import?

**Review by frozenbugs — APPROVED** [review]

File: python/dgl/__init__.py (hunk 2 above)
Comment (Rhett-Ying) [reply python/dgl/__init__.py:51]: I think it's better to update `dgl.ItemSet` in tests to `graphbolt.ItemSet` in this PR.

Comment (peizhou001) [reply python/dgl/__init__.py:51]: fixed

File: python/dgl/graphbolt/__init__.py (hunk 1 above)
Comment (peizhou001) [reply python/dgl/graphbolt/__init__.py]: Make sense, have dropped the changes.

**Pull request merged**